worlds/sc2wol/Locations.py

```diff
@@ -258,15 +271,19 @@ def get_locations(world: Optional[MultiWorld], player: Optional[int]) -> Tuple[L
                      lambda state: state._sc2wol_has_competent_comp(world, player)),
         LocationData("Shatter the Sky", "Shatter the Sky: Leviathan", SC2WOL_LOC_ID_OFFSET + 2805,
                      lambda state: state._sc2wol_has_competent_comp(world, player)),
-        LocationData("All-In", "All-In: Victory", None)
+        LocationData("All-In", "All-In: Victory", None,
+                     lambda state: state._sc2wol_final_mission_requirements(world, player))
     ]
 
     beat_events = []
 
-    for location_data in location_table:
+    for i, location_data in enumerate(location_table):
+        # Removing all item-based logic on No Logic
+        if logic_level == 2:
+            location_table[i] = location_data._replace(rule=lambda state: True)
+        # Generating Beat event locations
```

Comment:
```suggestion
    if logic_level == 2:
        def no_logic(state): 
            return True
    for i, location_data in enumerate(location_table):
        # Removing all item-based logic on No Logic
        if logic_level == 2:
            location_table[i] = location_data._replace(rule=no_logic)
        # Generating Beat event locations
```

This will use a bit less memory, as it's only one function instead of one per location